File: sample/ruby/main_map_sample.rb
require_relative 'build_box'
require_relative 'map_util'

room_name = "1000"
build_box = BuildBox.new(room_name)

build_box.set_box_size(0.1)
build_box.set_build_interval(0.001)

column_num, row_num = 257, 257
csv_file = 'map_38_138_100km.csv'
height_scale = 100
high_color = [0.5, 0, 0]
low_color = [0, 1, 0]
map_data = get_map_data_from_csv(csv_file, height_scale)
boxes = map_data['boxes']
max_height = map_data['max_height']
# skip = 1  # high power device
skip = 2  # normal device
# skip = 4  # low power device

(row_num / skip).times do |j|
  (column_num / skip).times do |i|
#     puts "#{i} #{j}"
    x = i
    z = j
    y = boxes[j * skip][i * skip]

    if y >= 0
      r, g, b = get_box_color(y, max_height, high_color, low_color)
#       print("r: #{r}, g: #{g}, b: #{b}\n")
      build_box.create_box(x, y, z, r, g, b, 1)
    end
  end
end

build_box.send_data()

File: sample/javascript/mainMapSample.mjs
import BuildBox from './buildBox.mjs';
import { getMapDataFromCSV, getBoxColor } from './mapUtil.mjs';

(async () => {
  const roomName = "1000";
  const buildBox = new BuildBox(roomName);

  buildBox.setBoxSize(0.1);
  buildBox.setBuildInterval(0.001);

  const columnNum = 257;
  const rowNum = 257;
  const csvFile = 'map_38_138_100km.csv';
  const heightScale = 100;
  const highColor = [0.5, 0, 0];
  const lowColor = [0, 1, 0];
  const mapData = getMapDataFromCSV(csvFile, heightScale);
  const boxes = mapData.boxes;
  const maxHeight = mapData.maxHeight;
  // const skip = 1;  // high power device
  const skip = 2;  // normal device
  // const skip = 4;  // low power device

  for (let j = 0; j < rowNum / skip; j++) {
    for (let i = 0; i < columnNum / skip; i++) {
      const x = i;
      const z = j;
      const y = boxes[j * skip][i * skip];
      const [r, g, b] = getBoxColor(y, maxHeight, highColor, lowColor);

      if (y >= 0) {
        buildBox.createBox(x, y, z, r, g, b, 1);
      }
    }
  }

  await buildBox.sendData();
})().catch(error => {
  console.error(error);
});
